modules/custom/apigee_kickstart_enhancement/src/Plugin/Condition/BlockType.php

```diff
@@ -0,0 +1,212 @@
+    ];
+
+    // Store the block uuid so that we can retrieve it on evaluate.
+    $form['uuid'] = [
+      '#type' => 'value',
+      '#value' => $block->uuid(),
+    ];
+
+    return parent::buildConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    $this->configuration['uuid'] = $form_state->getValue('uuid');
+    $this->configuration['bundles'] = array_filter($form_state->getValue('bundles'));
+    $this->configuration['is_adjacent'] = $form_state->getValue('is_adjacent');
+    parent::submitConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    if (empty($this->configuration['bundles']) && !$this->isNegated()) {
+      return TRUE;
+    }
+
+    $block_storage = $this->entityTypeManager->getStorage('block');
+    if (($uuid = $this->configuration['uuid']) && ($current_block = $block_storage->loadByProperties(['uuid' => $uuid]))) {
+      /** @var \Drupal\block\BlockInterface $current_block */
+      $current_block = reset($current_block);
+
+      // Find all blocks within the same region and theme.
+      /** @var \Drupal\block\BlockInterface[] $blocks */
+      if ($blocks = $block_storage->loadByProperties([
+        'theme' => $current_block->getTheme(),
+        'region' => $current_block->getRegion(),
+      ])) {
+        // Find all other visible blocks.
+        unset($blocks[$current_block->id()]);
+        $blocks = array_filter($blocks, function(BlockInterface $block) {
+          return $block->access('view');
+        });
+
+        if ($this->configuration['is_adjacent']) {
+          // Add the current block and sort the blocks by weight.
+          $blocks[$current_block->id()] = $current_block;
+          uasort($blocks, 'Drupal\block\Entity\Block::sort');
+
+          // Find adjacent blocks.
+          $block_ids = array_keys($blocks);
+          $key = (array_search($current_block->id(), $block_ids));
+          $blocks = array_slice($blocks, $key === 0 ? 0 : $key - 1, 2 + $key);
+          unset($blocks[$current_block->id()]);
+        }
+
+        // Check if we have a block content of configured type.
+        $block_content_storage = $this->entityTypeManager->getStorage('block_content');
+        foreach ($blocks as $block) {
+          if (($plugin = $block->getPlugin())
+            && ($plugin->getPluginDefinition()['provider'] == 'block_content')
+            && ($id = $this->uuidLookup->get($plugin->getDerivativeId()))
+            && ($block_content = $block_content_storage->load($id))) {
+            if (!empty($this->configuration['bundles'][$block_content->bundle()])) {
+              return FALSE;
+            }
+          }
+        }
+      }
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    // TODO: Implement summary() method.
```

Comment:
We should create a follow up for this so we don't forget about it.  It would be nice to have.